### DISCUSSION TITLE
Missing function

### DISCUSSION THREAD

**Discussion by laat0003** [gmdsi/GMDSI_notebooks#47] [Q&A] [answered]
Hi,

I think the function "extract_hds_arrays_and_list_dfs()" is missing from the pyemu.uitls.helpers.py file in the dependencies folder. I get an "EOF while searching" error when I reach that part of the fryberg_pstfrom_pest_setup notebook. I'm in the process of trying to adopt this workflow for one of my own projects. The project is already completed and submitted to the client so I'm using this as an exercise to see see the difference between a PESTPP-IES with regular pilot point + SEGLIST method (used in the submitted project) and the method demonstrated in this notebook.

Cheers,
Tariq

**Comment by rhugman** [comment, marked as answer]
Hey -

"extract_hds_arrays_and_list_dfs()" should be in ‘helpers.py’, in the pstfrom tutorial folder. Can you check?

Here:

https://github.com/gmdsi/GMDSI_notebooks/blob/main/tutorials/part2_1_pstfrom_pest_setup/helpers.py

**Reply by laat0003** [reply]
My bad, I'm a stooopid head. I thought it was the helpers.py file that was in the pyemu.utils folder. I didn't realize there was another file with the same name just sitting in the tutorial folder.
I didn't actually run the tutorial. Just read the code. Then I started adapting my own models flopy script to use the same methodology. I'll make sure to run the tutorials first next time. Cheers Rui, sorry to be a pain.
